### PR TITLE
Fix networking issues with swarm

### PR DESCRIPTION
**Pull request by ryardley**
Closes: #204 

- Ensure Kademlia is bootstrapping correctly
- Nodes are sending messages to one another correctly
- Nodes accept multiaddrs that resolve their DNS
- Nodes know not to retry dialing themselves if dialing fails because the multiaddr is the same as the node
- We can upgrade individual containers using `docker service update --force`
- Documentation

## Summary by CodeRabbit

- **New Features**
	- Introduced new configuration files for deployment, including `agg.yaml`, `cn1.yaml`, `cn2.yaml`, and `cn3.yaml`.
	- Added scripts for building images, deploying stacks, copying secrets, and inspecting logs by service version.
	- Implemented a logging mechanism for network services and improved event handling in tests.
	- Added a new environment configuration file `.env.example` for blockchain application settings.
	- Introduced a new JSON file `example.secrets.json` for secret management.
	- Enhanced the Docker Compose setup with a new configuration file and multiple services.

- **Bug Fixes**
	- Enhanced error handling and logging in peer-to-peer message handling.

- **Documentation**
	- Updated deployment instructions in `swarm_deployment.md` and added a new example secrets file.

- **Chores**
	- Removed outdated Docker Compose configurations and scripts.

### CODE REVIEW

**Comment by ryardley** [comment]
Docker swarm compatibility appears to be working over here: https://github.com/ryardley/libp2p-kad-gossipsub-quic-example/tree/main next steps would be to integrate this in here

**Comment by ryardley** [comment]
Leaving this for today. 

Current status:

```
docker service logs --raw enclave_cn3
Setting password
invalid string length for key "default.address" in .config/enclave/config.yaml YAML file
Setting password
invalid string length for key "default.address" in .config/enclave/config.yaml YAML file
Setting password
invalid string length for key "default.address" in .config/enclave/config.yaml YAML file
Setting password
invalid string length for key "default.address" in .config/enclave/config.yaml YAML file
```

Need to check what is happening with the env substitution which is obviously failing.

**Comment by ryardley** [comment]
So I need swarm round robin in order to allow services to connect to one another but I need to expose ports for external nodes to connect to the services - these two settings are unfortunately not compatible. 😭

**Comment by ryardley** [comment]
> Local swarm deployment works BUT I need swarm round robin in order to allow services to connect to one another but I need to expose ports for external nodes to connect to the services - these two settings are unfortunately not compatible. 😭
> 
> @hmzakhalid keen to work through some of this with you. I am wondering about adding a load balancer to connect to nodes however it seems like libp2p doesn't really like to work behind a load balancer (hence round robin dns)

I worked this out by using `mode: host`

**Comment by ryardley** [comment]
I believe the last piece of this is to convert the current 'net' test to use swarm so we accurately test the networking ability of the package

**Comment by ryardley** [comment]
I believe the last piece of this is to convert the current 'net' test to use swarm so we accurately test the networking ability of the package

File: packages/ciphernode/net/src/dialer.rs [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    peers: &Vec<String>,
+) -> Result<()> {
+    let futures: Vec<_> = peers
+        .iter()
+        .map(|addr| dial_multiaddr(cmd_tx, event_tx, addr))
+        .collect();
+    let results = join_all(futures).await;
+    results.into_iter().for_each(trace_error);
+    Ok(())
+}
```
[the remark this answers is not in the document]
Comment (ryardley) [reply packages/ciphernode/net/src/dialer.rs:60]: Feel like this is overkill

File: packages/ciphernode/net/src/network_manager.rs [hunk 2]
```diff
@@ -9,15 +12,18 @@ use enclave_core::{EnclaveEvent, EventBus, EventId, Subscribe};
...
 use tracing::{error, info, instrument, trace};
 
 /// NetworkManager Actor converts between EventBus events and Libp2p events forwarding them to a
 /// NetworkPeer for propagation over the p2p network
 pub struct NetworkManager {
     bus: Addr<EventBus>,
-    tx: Sender<Vec<u8>>,
+    tx: mpsc::Sender<NetworkPeerCommand>,
     sent_events: HashSet<EventId>,
+    topic: String,
```
[the remark this answers is not in the document]
Comment (ryardley) [reply packages/ciphernode/net/src/network_manager.rs:26]: This is a good idea but not now.

**Comment by ryardley** [comment]
We should be good to go with this. @hmzakhalid can you please check that nodes are connecting and everything is working with the merge to release.

**Comment by hmzakhalid** [comment]
> We should be good to go with this. @hmzakhalid can you please check that nodes are connecting and everything is working with the merge to release.

On it!

**Pull request merged**